watertrading/code/Visualization.py

```diff
@@ -0,0 +1,919 @@
+    demand_pivot = pd.pivot_table(df_profits_demand[["Operator","Profit"]], index="Operator", values="Profit", aggfunc="sum")
+    midstream_pivot = pd.pivot_table(df_profits_midstream_transport[["Operator","Profit"]], index="Operator", values="Profit", aggfunc="sum")
+    p_transport_pivot = pd.pivot_table(df_profits_producer_transport[["Operator","Profit"]], index="Operator", values="Profit", aggfunc="sum")
+
+    # Create figure and axes handles
+    fig = plt.figure(figsize=(8,8))
+    ax1 = plt.subplot(3,1,1)
+    ax2 = plt.subplot(3,1,2)
+    ax3 = plt.subplot(3,2,5)
+    ax4 = plt.subplot(3,2,6)
+    ax1.bar(x = [ind for ind,row in supply_pivot.iterrows()], height = supply_pivot["Profit"].to_list(), color = "red")
+    ax1.set_ylim(GetLims(supply_pivot["Profit"].to_list()))
+    ax1.tick_params(axis='x', labelrotation=45)
+    ax2.bar(x = [ind for ind,row in demand_pivot.iterrows()], height = demand_pivot["Profit"].to_list(), color = "blue")
+    ax2.set_ylim(GetLims(demand_pivot["Profit"].to_list()))
+    ax2.tick_params(axis='x', labelrotation=45)
+    ax3.bar(x = [ind for ind,row in midstream_pivot.iterrows()], height = midstream_pivot["Profit"].tolist(), color = "gold")
+    ax3.set_ylim(GetLims(midstream_pivot["Profit"].to_list()))
+    ax3.tick_params(axis='x', labelrotation=45)
+    ax4.bar(x = [ind for ind,row in p_transport_pivot.iterrows()], height = p_transport_pivot["Profit"].tolist(), color = "orange")
+    ax4.set_ylim(GetLims(p_transport_pivot["Profit"].to_list()))
+    ax4.tick_params(axis='x', labelrotation=45)
+    ax1.set(xlabel="Producer ID", ylabel="Surplus [USD]")
+    ax2.set(xlabel="Consumer ID", ylabel="Surplus [USD]")
+    ax3.set(xlabel="Midstream ID", ylabel="Surplus [USD]")
+    ax4.set(xlabel="Producer ID", ylabel="Surplus [USD]")
+    ax1.set_title(f"Supplier Surplus", ha='left', x=-0)
+    ax2.set_title(f"Consumer Profit", ha='left', x=-0)
+    ax3.set_title(f"Midstream Profit", ha='left', x=-0)
+    ax4.set_title(f"Supplier Transportation Profit", ha='left', x=-0)
+    plt.tight_layout(pad=1.08, h_pad=None, w_pad=None, rect=(0,0,1,1))
+
+    # save figure and return plot handle
+    plt.savefig(os.path.join(input_dir,fig_name))
+    return fig
+
+
+##################################################
+def RevenuePlot(input_dir,MATCHES,fig_name="plot_revenue"):
+    """
+    Creates a plot of stakeholder revenue streams
+    """
+    df_matches_supply, df_matches_demand, df_matches_producer_transport, df_matches_midstream_transport = fetch_match_data(MATCHES)
+
+    # Create figure and axes handles
+    fig = plt.figure(figsize=(6.4,8))
+    ax1 = plt.subplot(3,1,1)
+    ax2 = plt.subplot(3,1,2)
+    ax3 = plt.subplot(3,2,5)
+    ax4 = plt.subplot(3,2,6)
+    # NOTE: I am using the -1 in the below so that the resulting graph gives the expecgted visual interpretation of revenue, instead of the economic one
```

Comment:
```suggestion
    # NOTE: I am using the -1 in the below so that the resulting graph gives the expected visual interpretation of revenue, instead of the economic one
```

Reply:
Correcgted